packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdk.kt

```diff
@@ -285,6 +304,44 @@ class DdSdk(
         }
     }
 
+    private fun buildVitalUpdateFrequency(vitalsUpdateFrequency: String?): VitalsUpdateFrequency {
+        val vitalUpdateFrequencyLower = vitalsUpdateFrequency?.lowercase(Locale.US)
+        return when (vitalUpdateFrequencyLower) {
+            "never" -> VitalsUpdateFrequency.NEVER
+            "rare" -> VitalsUpdateFrequency.RARE
+            "average" -> VitalsUpdateFrequency.AVERAGE
+            "frequent" -> VitalsUpdateFrequency.FREQUENT
+            else -> VitalsUpdateFrequency.AVERAGE
+        }
+    }
+
+    private fun monitorJsRefreshRate(vitalsUpdateFrequency: VitalsUpdateFrequency) {
+        val frameTimeCallback = buildFrameTimeCallback(vitalsUpdateFrequency)
+        reactContext.runOnJSQueueThread {
+            val vitalFrameCallback =
+                VitalFrameCallback(frameTimeCallback) { isInitialized() }
+            try {
+                Choreographer.getInstance().postFrameCallback(vitalFrameCallback)
+            } catch (e: IllegalStateException) {
+                // This should never happen as the React Native thread always has a Looper
+                val errorMessage = e.message
+                datadog.telemetryError(errorMessage ?: MONITOR_JS_ERROR_KIND, e.stackTraceToString(), MONITOR_JS_ERROR_KIND)
+            }
+        }
+    }
+
+    private fun buildFrameTimeCallback(vitalsUpdateFrequency: VitalsUpdateFrequency): (frameTime: Double) -> Unit {
+        val monitorJsRefreshRate = vitalsUpdateFrequency != VitalsUpdateFrequency.NEVER
+        return {
+            if (monitorJsRefreshRate && it > 0.0) {
```

Comment:
just curious: is it possible to have negative `frameTime`?

Reply:
I had a previous comment in the PR for this TODO, and I think we could remove it completely:

I'm not sure it's worth logging this error.
It would require to inject a error handler into the VitalFrameCallback, making the code more complex that I think it should.
I think in theory it should not happen if we've been able to start the Frame Callback from the SDK where the error is handled.

If you think I should log the error I will pass a error handler to the VitalFrameCallback contructor.

Reply:
Good question, I could not find the answer. The big issue would not so much be a negative `frameTime` but a `0.0` frametime that would crash the app.
I think it's still worth leaving it here as it does not add a lot of overhead.